drafts/25-xxx-specifications.txt

```diff
@@ -206,64 +206,348 @@ to11. The following tokens also appear in the illustrative syntax below.
+
+Example syntax:
+      #define ID() replacement-list EOL
+      #define ID(identifier-list) replacement-list EOL
+      #define ID(...) replacement-list EOL
+      #define ID(identifier-list, ...) replacement-list EOL
+
+
+3.2.1 Static semantics specifications
+--------------------------------------
+
+df02. The identifier ID must be immediately followed by the
+      left parenthesis '(' with no intervening whitespace.
+
+df04. The identifier-list is a comma-separated list of ID tokens.
+
+df06. No identifier may appear more than once in the identifier-list.
+
+df08. The identifier names in the identifier-list have "scope" in the
+      replacement list. (See the section on macro expansion for the
+      semantics of function-like macro expansion.)
+
+df10. The replacement-list may be the empty sequence of tokens.
+
+df12. Whitespace before or after the replacement-list is not
+      considered to be part of the replacement-list.
+
+df16. The "..." between the parentheses specifies that the function-like
+      macro may be invoked with a variable number of arguments. (See the
+      section on macro expansion for the semantics of function-like macros
+      with a variable number of arguments.)
+
+
+3.2.2 Evaluation semantics specifications
+-----------------------------------------
+
+df30. This #define directive defines a function-like macro with parameters
+      (dummy arguments) as specified in the identifier-list and function
+      body as specified in the replacement list.
+
+df32. This #define directive does not scan the replacement-list
+      for macros to expand.
+
+df34. An identifier currently defined as a function-like macro shall not
+      be redefined by another #define preprocessing directive unless
+      the second definition is a function-like macro definition and
+      the two replacement lists are identical (as in C 2023).
+
 
 3.3 The '#undef' directive
 --------------------------
 
+Example syntax:
+      #undef ID EOL
+
+
+3.3.1 Static semantics specifications
+-------------------------------------
+
+ud02. ID shall not be one of the macros defined in the "Predefined macros"
+      section below.
+
+ud04. The specified identifier may or may not have been defined.
 
 
+3.3.2 Evaluation semantics specifications
+-----------------------------------------
+
+ud20. If no definition exists for the identifier ID, this directive has
+      no effect.
+
+ud22. The definition for the object-like macro or function-like macro
+      named by ID is removed.
+
+ud24.
+
 3.4 The '#include' directive
 ----------------------------
 
+Example syntax:
+      #include CHARACTER_STRING EOL
+      #include <character-list> EOL
+
+
+3.4.1 Static semantics specifications
+--------------------------------------
+
+in02. In the first form, the CHARACTER_STRING may be delimited with
+      single-quote "'" or double-quote '"' characters.
+
+in04. A "_kind" specifier shall not appear following the CHARACTER_STRING.
```

Comment:
According to F23, the optional kind type parameter actually comes _before_ a character literal constant. However the positioning is also an irrelevant detail.

```suggestion
in04. CHARACTER_STRING shall not include a kind type parameter.
```

However at this point it might just be easier to simply redefine `CHARACTER_STRING`:

```
 | CHARACTER_STRING    | RegEx "[^"\n]*"  |
```
which allows us to delete this rule and the one above it.

Reply:
In today's meeting we resolved to define the `CHARACTER_STRING` token as suggested above, and adjust all references accordingly.